### ISSUE TITLE
cannot open '/01-main.repo'

### ISSUE THREAD

**Issue by Asarox84** [wimpysworld/deb-get#830]
### Expected behavior

Not seeing the errors below I presume.
Variable ETC_DIR replacement seems to be broken [here](https://github.com/wimpysworld/deb-get/blob/main/deb-get#L714)

### Actual behavior

Running update gives : 
```
tail: cannot open '/01-main.repo' for reading: No such file or directory
No URLs found in /01-main.repo.tmp.
```

### Steps to reproduce

sudo deb-get update :

```
 /etc/deb-get  sudo deb-get update
  [+] Updating /etc/deb-get/01-main
tail: cannot open '/01-main.repo' for reading: No such file or directory
No URLs found in /01-main.repo.tmp.
  [+] Updating cache of supported apps in the background
...
Reading package lists...
```

### System information

Run the following a paste the output below:

#### /etc/os-release

```NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

#### /etc/lsb-release

```DISTRIB_ID=Ubuntu
DISTRIB_RELEASE=20.04
DISTRIB_CODENAME=focal
DISTRIB_DESCRIPTION="Ubuntu 20.04.6 LTS"
```

#### deb-get version

`0.3.8`

**Comment by philclifford** [comment]
Do not sudo deb-get.
If you try again without sudo does it still fail ?

**Comment by Asarox84** [comment]
Here it is : 

```
 ~  deb-get update
[sudo] password for Matthieu.Meraville: 
  [+] Updating /etc/deb-get/01-main
tail: cannot open '/01-main.repo' for reading: No such file or directory
No URLs found in /01-main.repo.tmp.
  [+] Updating cache of supported apps in the background
  [+] Updating /var/cache/deb-get/deb-get.json
  [!] ERROR! Missing required information of github package deb-get:
URL=
VERSION_PUBLISHED=
cut: /var/cache/deb-get/supported.list.tmp: Permission denied
  [!] ERROR! Missing required information of github package deb-get:
URL=
VERSION_PUBLISHED=
```

**Comment by philclifford** [comment]
Thanks. 🤔

**Comment by philclifford** [comment]
I cannot replicate this behaviour in a focal VM.  I installed `deb-get` 0.3.8 per the readme, updated, installed a few apps - all good.  Then did a `sudo deb-get update` and got some expected errors (empty cache files because root doesn't get the API key ...) but not the behaviour seen above. Subsequent `deb-get update` works normally (after `deb-get clean` to remove any bad cache files the `sudo deb-get update` had created).  

Was yours a fresh install or an update from a previous version?  If an update, from which version and how did you do the update? Did you have a working `deb-get` before this occurred ?

You need to remove any `/01-main.repo*` junk, make sure you have no hung old `deb-get`s running and probably best to  

```bash
sudo rm /var/cache/deb-get/*
``` 

too.  If that doesn't get `deb-get update` to work normally I'd be thinking keep a copy of `/etc/deb-get/installed` somewhere then purge the old `deb-get` install and start over, but I'd like to understand how and why your `deb-get` is broken. :thinking: :shrug:

**Comment by Asarox84** [comment]
Hello,

I even tried with the deb-get clean, then deb-get clean (and sudo deb-get clean) and then rm sudo rm /var/cache/deb-get/* and then reinstall everything, I still have issues :

```
deb-get list
1password
agena
  [*] WARNING! Cached file /var/cache/deb-get/android-messages-desktop.json is empty or missing.
android-messages-desktop
  [*] WARNING! Cached file /var/cache/deb-get/antimicrox.json is empty or missing.
antimicrox
anydesk
  [*] WARNING! Cached file /var/cache/deb-get/appimagelauncher.json is empty or missing.
appimagelauncher
atom
atomic
audio-recorder
  [*] WARNING! Cached file /var/cache/deb-get/authme.json is empty or missing.
authme
azure-cli
  [*] WARNING! Cached file /var/cache/deb-get/azuredatastudio.json is empty or missing.
azuredatastudio
  [*] WARNING! Cached file /var/cache/deb-get/balena-etcher-electron.json is empty or missing.
balena-etcher-electron
  [*] WARNING! Cached file /var/cache/deb-get/bat.json is empty or missing.
bat
  [*] WARNING! Cached file /var/cache/deb-get/battery-monitor.json is empty or missing.
battery-monitor
beersmith3
  [*] WARNING! Cached file /var/cache/deb-get/bitwarden.json is empty or missing.
bitwarden
bitwig-studio
blanket
  [*] WARNING! Cached file /var/cache/deb-get/blockbench.json is empty or missing.
blockbench
bluejeans-v2
  [*] WARNING! Cached file /var/cache/deb-get/bottom.json is empty or missing.
bottom
brave-browser
  [*] WARNING! Cached file /var/cache/deb-get/brisqi.json is empty or missing.
brisqi
  [*] WARNING! Cached file /var/cache/deb-get/caprine.json is empty or missing.
caprine
cawbird
chronograf
code
  [*] WARNING! Cached file /var/cache/deb-get/codium.json is empty or missing.
codium
  [*] WARNING! Cached file /var/cache/deb-get/com.github.tkashkin.gamehub.json is empty or missing.
com.github.tkashkin.gamehub
copyq
  [*] WARNING! Cached file /var/cache/deb-get/cosign.json is empty or missing.
cosign
  [*] WARNING! Cached file /var/cache/deb-get/croc.json is empty or missing.
croc
cronopete
crossover
cryptomator
dbeaver-ce
deadbeef-static
  [*] WARNING! Cached file /var/cache/deb-get/deb-get.json is empty or missing.
deb-get                         [ installed ]
  [*] WARNING! Cached file /var/cache/deb-get/deborah.json is empty or missing.
deborah
deltachat-desktop
  [*] WARNING! Cached file /var/cache/deb-get/deskcut.json is empty or missing.
deskcut
  [!] ERROR! Missing required information of direct package discord:
URL=https://discord.com/api/download?platform=linux&format=deb
VERSION_PUBLISHED=

```

**Comment by Asarox84** [comment]
I removed everything again then : 

```
$ curl -sL https://raw.githubusercontent.com/wimpysworld/deb-get/main/deb-get | sudo -E bash -s install deb-get
  [+] Updating /var/cache/deb-get/deb-get.json
/var/cache/deb-get/deb-get_0. 100%[=================================================>]  22,29K  --.-KB/s    in 0,02s   
Selecting previously unselected package deb-get.
(Reading database ... 381818 files and directories currently installed.)
Preparing to unpack .../deb-get_0.3.8-1_all.deb ...
Unpacking deb-get (0.3.8-1) ...
Setting up deb-get (0.3.8-1) ...
  [+] Updating /etc/deb-get/01-main
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100  108k  100  108k    0     0  21520      0  0:00:05  0:00:05 --:--:-- 40907
  [+] Updating cache of supported apps in the background
Processing triggers for man-db (2.9.1-1) ...
```

then 

```
$ deb-get list
deb-get                         [ installed ]
```

then 

```
$ deb-get update
  [+] Updating /etc/deb-get/01-main
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:02 --:--:--     0
100  108k  100  108k    0     0  21625      0  0:00:05  0:00:05 --:--:-- 84428
  [+] Updating cache of supported apps in the background
  [!] ERROR! Missing required information of github package deb-get:
URL=
VERSION_PUBLISHED=
xxxxx.yyyyyy:~$   [!] ERROR! Missing required information of website package agena:
URL=
VERSION_PUBLISHED=
cut: /var/cache/deb-get/supported.list.tmp: Permission denied
```

I thought it was because I'm under zsh by default but I launched everything under bash for the test.

**Comment by philclifford** [comment]
Well that's looking better - not right yet but ... 

```
cut: /var/cache/deb-get/supported.list.tmp: Permission denied
```

this should not happen: ownership and permissions on `var/cache/deb-get`  should be root:root 755 and the file should be 644.  If the directory permissions are too restrictive or the umask too restrictive that would account for the above breakage.  It would also account for "you" not being able to read the cache files so getting the missing information errors on `update` I don't think your shell being `zsh` is the issue as `deb-get` should be running under `/usr/bin/env bash` but if that isn't bash v5.x it could be a problem.

**Comment by philclifford** [comment]
That's the problem then: root default umask is 0022 : 
```shell
$ sudo bash -c "umask"
0022
```
 ( and for users 0002).  You have hardened that somewhere, and that has broken `deb-get`.  
You can fix by either ensuring root's effective umask under `sudo` permits reading or edit `/usr/bin/deb-get` and add `${ELEVATE} ` to the failing lines (anything reading in `${CACHE_DIR}`  e.g. `cut` `cat` `grep` `test` ....  ) (and be prepared to patch every update ...)

**Comment by yermulnik** [comment]
Found myself in the same posture:
```shell
tail: cannot open '/01-main.repo' for reading: No such file or directory
```
The `/etc/deb-get/01-main.repo` file was empty. Deleting it solved the issue.

**Comment by miljanuscumlic** [comment]
I managed to reproduce this error. When deb-get command is invoked without the network connection, `/etc/deb-get/01-main.repo ` becomes empty. Deleting this file as @yermulnik said, solves the problem.

**Comment by JohnPettigrew** [comment]
I've had the same error, and I think I share the same cause @miljanuscumlic mentioned - I had a transient loss of web connection, and then deb-get started producing this error. Deleting the file restored expected behaviour.

**Comment by philclifford** [comment]
Thanks @yermulnik, @miljanuscumlic and @JohnPettigrew . Sorry it took so long to notice the separate issue that was the source of your experiences.  That should be fixed by #968
